### DISCUSSION TITLE
Titlebar of the focused window disappears when exec'ing krunner

### DISCUSSION THREAD

**Discussion by l1yefeng** [i3/i3#4655] [Q&A] [answered]
Hi, I need help to solve a small issue: every time I start KRunner, the titlebar of the focused window disappears, and doesn't come back after KRunner hides. KRunner is the launcher of KDE, similar to Rofi.

Here is my computer's setup:

```
OS: Debian GNU/Linux 11 (bullseye) x86_64
DE: KDE Plasma 5.20.5
WM: i3
```

Relevant part of i3/config (all the rest are `bindsym`'s):

```
for_window [class=".*"] title_format " %title"

for_window [title="Desktop — Plasma"] kill; floating enable; border none
for_window [class="plasmashell"] floating enable
for_window [class="Plasma"] floating enable; border none
for_window [title="plasma-desktop"] floating enable; border none
for_window [title="win7"] floating enable; border none
for_window [class="krunner"] floating enable; border none
for_window [class="Kmix"] floating enable; border none
for_window [class="Klipper"] floating enable; border none
for_window [class="Plasmoidviewer"] floating enable; border none
for_window [class="(?i)*nextcloud*"] floating disable
for_window [class="plasmashell" window_type="notification"] floating enable, border none, move right 700px, move down 450px
no_focus [class="plasmashell" window_type="notification"]

set $menu --no-startup-id qdbus org.kde.krunner /App display
bindsym $mod+d exec $menu
```

Thanks for reading.

**Comment by Airblader** [comment, marked as answer]
A command of the structure

```
[<criteria>] command1; command2
```

means "execute command1 on the matched windows, and command2 on the focused window". You probably want to use

```
[<criteria>] command1, command2
```

which means "execute command1 and then command2 on all matched windows".

In other words, using `;` to chain commands breaks the relationship with the match criteria while `,` preserves it.

**Reply by l1yefeng** [reply]
Thanks a lot. I copied it from KDE wiki and didn't suspect it a bit. You are right.
